hudi-common/src/main/java/org/apache/hudi/metadata/SecondaryIndexRecordGenerationUtils.java

```diff
@@ -207,19 +226,8 @@ private static Map<String, String> getRecordKeyToSecondaryKey(HoodieTableMetaCli
     if (dataFilePath.isPresent()) {
       baseFileReader = Option.of(HoodieIOFactory.getIOFactory(metaClient.getStorage()).getReaderFactory(recordMerger.getRecordType()).getFileReader(getReaderConfigs(storageConf), dataFilePath.get()));
```

Comment:
This reader is never getting closed and it will lead to leaks.

Reply:
Yes.  This is fixed now.